### PR TITLE
Add missing headers to azure-attestation\README.md

### PR DESCRIPTION
**Pull request by mwynne**
# Description

Added sections per #4554. This is a work in progress. 

 @scbedd before I get really invested with the content under the new headers, I just wanted to make sure I'm going in the right direction with this. It's a pretty old issue, and I noticed that a few of the links in the original issue are broken and it appears there has been some refactoring in the repo since. Thanks.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by mwynne** [comment]
@microsoft-github-policy-service agree

File: sdk/attestation/azure-mgmt-attestation/README.md [hunk 1]
```diff
@@ -8,21 +8,80 @@ For a more complete view of Azure libraries, see the [azure sdk python release](
...
 
+
+### Resources
 To learn how to use this package, see the [quickstart guide](https://aka.ms/azsdk/python/mgmt)
  
 For docs and references, see [Python SDK References](https://docs.microsoft.com/python/api/overview/azure/)
 Code samples for this package can be found at [Attestation Management](https://docs.microsoft.com/samples/browse/?languages=python&term=Getting%20started%20-%20Managing&terms=Getting%20started%20-%20Managing) on docs.microsoft.com.
 Additional code samples for different Azure services are available at [Samples Repo](https://aka.ms/azsdk/python/mgmt/samples)
 
+## Key Concepts
```
Comment (scbedd) [inline sdk/attestation/azure-mgmt-attestation/README.md:32]:
```suggestion
```

File: sdk/attestation/azure-mgmt-attestation/README.md [hunk 2]
```diff
@@ -8,21 +8,80 @@ For a more complete view of Azure libraries, see the [azure sdk python release](
...
+Content TBD
```
Comment (scbedd) [inline sdk/attestation/azure-mgmt-attestation/README.md:33]:
```suggestion
```

File: sdk/attestation/azure-mgmt-attestation/README.md [hunk 3]
```diff
@@ -8,21 +8,80 @@ For a more complete view of Azure libraries, see the [azure sdk python release](
...
+
+client = AttestationManagementClient(
+    credential=DefaultAzureCredential(),
+    subscription_id="00000000-0000-0000-0000-000000000000",
+)
+
+response = client.attestation_providers.list()
+print(response)
+```
+TBC
```
Comment (scbedd) [inline sdk/attestation/azure-mgmt-attestation/README.md:51]:
```suggestion
```

File: sdk/attestation/azure-mgmt-attestation/README.md [hunk 4]
```diff
@@ -8,21 +8,80 @@ For a more complete view of Azure libraries, see the [azure sdk python release](
...
+
+## Troubleshooting
+
+### CredentialUnavailableError
+The CredentialUnavailableError is a specific error type derived from ClientAuthenticationError. This error type is used to indicate that the credential can't authenticate in the current environment, due to missing required configuration or setup. This error is also used as an indication for chained credential types, such as DefaultAzureCredential and ChainedTokenCredential, that the chained credential should continue to attempt other credential types later in the chain.
 
-# Provide Feedback
+### Permission issues
+Service client calls that result in HttpResponseError with a StatusCode of 401 or 403 often indicate the caller doesn't have sufficient permissions for the specified API. Check the service documentation to determine which RBAC roles are needed for the specific request, and ensure the authenticated user or service principal have been granted the appropriate roles on the resource.
```
Comment (scbedd) [inline sdk/attestation/azure-mgmt-attestation/README.md:59]:
```suggestion
Service client calls that result in HttpResponseError with a StatusCode of 401 or 403 often indicate the caller doesn't have sufficient permissions for the specified API. Check the [service documentation](https://learn.microsoft.com/en-us/azure/attestation/) to determine which RBAC roles are needed for the specific request, and ensure the authenticated user or service principal have been granted the appropriate roles on the resource.
```

**Review by scbedd — CHANGES_REQUESTED** [review]
Few nits cleaning up TBC/TBD and adding a link to the service documentation.

**Comment by mwynne** [comment]
@scbedd I'm glad I checked before moving ahead with anything else! I addressed your comments in a new commit. Let me know if I missed anything, or if you'd like additional changes. 

I'll go find something else to work on. Since I'm new around here, if there are any issues that no one else wants to volunteer for that you may want to suggest, please let me know.

**Review by scbedd — APPROVED** [review]
Thank you for the contribution @mwynne ! I'm going to ignore this error (which has nothing to do with your PR) and get this merged.

**Comment by scbedd** [comment]
/check-enforcer override

Overriding as the `mypy` error is most definitively not from this PR.

**Pull request merged**